std/traits.d

```diff
@@ -4125,6 +4125,13 @@ unittest
     static assert(is(ImplicitConversionTargets!(string)[0] == const(char)[]));
 }
 
+unittest // bugzilla 15940
+{
+    class Foo{}
+    struct Bar{Foo foo; alias foo this;}
+    static assert(ImplicitConversionTargets!(Bar).length == 0);
```

Comment:
> it looks like ImplicitConversionTargets doesn't recognize alias this at all.

Yes, it's was just working for a struct containing an alias this to a class (so the _bug_ fixed here), but all the other cases were not. for example:

```D
unittest // bugzilla 15940
{
    struct Bar{ubyte d; alias d this;}
    static assert(ImplicitConversionTargets!(Bar).length == 0);
}
 ```
passes. So the fix is conform will the global behavior of the template.